src/lib/configs/project/dockerCompose.ts

```diff
@@ -0,0 +1,315 @@
+  name: string;
+  tcpPort: number;
+  webSocketPort: number;
+  bootStrapNoxName: string;
+};
+
+function genNox({
+  name,
+  tcpPort,
+  webSocketPort,
+  bootStrapNoxName,
+}: GenNoxImageArgs): [name: string, service: Service] {
+  return [
+    name,
+    {
+      image: versions.nox,
+      pull_policy: "always",
+      ports: [`${tcpPort}:${tcpPort}`, `${webSocketPort}:${webSocketPort}`],
+      environment: {
+        FLUENCE_ENV_AQUA_IPFS_EXTERNAL_API_MULTIADDR,
+        FLUENCE_ENV_DECIDER_IPFS_MULTIADDR: NOX_IPFS_MULTIADDR,
+        FLUENCE_ENV_AQUA_IPFS_LOCAL_API_MULTIADDR: NOX_IPFS_MULTIADDR,
+        FLUENCE_ENV_CONNECTOR_API_ENDPOINT: `http://${CHAIN_CONTAINER_NAME}:${CHAIN_PORT}`,
+        FLUENCE_ENV_CONNECTOR_FROM_BLOCK: "earliest",
+        WASM_LOG: "info",
+        RUST_LOG:
+          "debug,particle_reap=debug,aquamarine=warn,aquamarine::particle_functions=debug,aquamarine::log=debug,aquamarine::aqua_runtime=error,ipfs_effector=off,ipfs_pure=off,system_services=debug,marine_core::module::marine_module=info,tokio_threadpool=info,tokio_reactor=info,mio=info,tokio_io=info,soketto=info,yamux=info,multistream_select=info,libp2p_secio=info,libp2p_websocket::framed=info,libp2p_ping=info,libp2p_core::upgrade::apply=info,libp2p_kad::kbucket=info,cranelift_codegen=info,wasmer_wasi=info,cranelift_codegen=info,wasmer_wasi=info,run-console=trace,wasmtime_cranelift=off,wasmtime_jit=off,libp2p_tcp=off,libp2p_swarm=off,particle_protocol::libp2p_protocol::upgrade=info,libp2p_mplex=off,particle_reap=off,netlink_proto=warn",
+        FLUENCE_SYSTEM_SERVICES__ENABLE: "aqua-ipfs,decider",
+        FLUENCE_ENV_CONNECTOR_WALLET_KEY:
+          "0xfdc4ba94809c7930fe4676b7d845cbf8fa5c1beae8744d959530e5073004cf3f",
+        FLUENCE_ENV_CONNECTOR_CONTRACT_ADDRESS:
+          "0x0f68c702dC151D07038fA40ab3Ed1f9b8BAC2981",
+        FLUENCE_SYSTEM_SERVICES__DECIDER__DECIDER_PERIOD_SEC: 10,
+        FLUENCE_MAX_SPELL_PARTICLE_TTL: "9s",
+        FLUENCE_SYSTEM_SERVICES__DECIDER__NETWORK_ID: 31337,
+        FLUENCE_ROOT_KEY_PAIR__PATH: `/run/secrets/${name}`,
+      },
+      command: [
+        "--aqua-pool-size=5",
+        `-t=${tcpPort}`,
+        `-w=${webSocketPort}`,
+        "--external-maddrs",
+        `/dns4/${name}/tcp/${tcpPort}`,
+        `/dns4/${name}/tcp/${webSocketPort}/ws`,
+        "--allow-private-ips",
+        tcpPort === TCP_PORT_START
+          ? "--local"
+          : `--bootstraps=/dns/${bootStrapNoxName}/tcp/${TCP_PORT_START}`,
+      ],
+      depends_on: [IPFS_CONTAINER_NAME],
+      secrets: [name],
+    },
+  ];
+}
+
+async function genDockerCompose(
+  providerConfig: ProviderConfigReadonly,
+): Promise<LatestConfig> {
+  const secretKeys = await genSecretKeysOrReturnExisting(
+    Object.keys(providerConfig.computePeers),
+  );
+
+  const [bootStrapNoxNameAndKey, ...restSecretKeys] = secretKeys;
+
+  assert(
+    bootStrapNoxNameAndKey !== undefined,
+    "At least one nox is required to generate a docker-compose.yml",
```

Comment:
for local network you will need to have at least one nox. The fact that it will be bootstrap shouldn't matter. This error should never happen it's mostly a typescript pleaser because in TS there is no easiily available type "NonEmptyArray". I validate the config beforehand

Reply:
it's just strange that you check against `bootStrapNoxNameAndKey`, not against number of noxes user requested

Reply:
It's because of the typescript. I have to convince it I checked the first element in the array actually exists. It can't infer this idea from just checking the array length before that